### PR TITLE
[SPARK-15442][ML][PYSPARK] Add 'relativeError' param to PySpark QuantileDiscretizer

### PR DESCRIPTION
**Pull request by MLnick**
This PR adds the `relativeError` param to PySpark's `QuantileDiscretizer` to match Scala.

Also cleaned up a duplication of `numBuckets` where the param is both a class and instance attribute (I removed the instance attr to match the style of params throughout `ml`).

Finally, cleaned up the docs for `QuantileDiscretizer` to reflect that it now uses `approxQuantile`.
## How was this patch tested?

A little doctest and built API docs locally to check HTML doc generation.

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -1250,6 +1257,20 @@ def getNumBuckets(self):
...
+    def setRelativeError(self, value):
+        """
+        Sets the value of :py:attr:`relativeError`.
+        """
+        return self._set(relativeError=value)
+
+    @since("2.0.0")
+    def getRelativeError(self):
+        """
+        Gets the value of relativeError or its default value.
```
Comment (BryanCutler) [inline python/pyspark/ml/feature.py]: no biggie, but the setter has a link for the param :py:attr:`relativeError`

**Comment by BryanCutler** [comment]
LGTM

**Comment by jkbradley** [comment]
Is this targeted at 2.0?  I'd prefer we focus on finishing documentation & QA checks before working on new features and APIs.

**Comment by MLnick** [comment]
@jkbradley this is a missing param on the python side - I noticed it while
reviewing the PR for doc for count vectorizer, hashing tf and quantile
discretizer.

It also contains doc fixes as quantile discretizer doc is incorrect now
that it uses approx quantile.

I thought we were still trying to get as much of pyspark parity for 2.0 as
we can?

Otherwise I can just make the doc changes

**Comment by jkbradley** [comment]
I do want us to push for pyspark parity as much as possible, but I made those QA JIRAs for Python parity checks to do an audit, not to achieve parity for 2.0.  I definitely find it tempting to push as much API parity stuff in as possible, but I've gradually become convinced it's better to just focus on QA & docs to get the release out the door.  As soon as the release is done, we can focus on the newly created Python JIRAs and other new APIs, etc.

But it's your call on this one.  If this is ready, then maybe it's better to just go ahead and merge it to avoid wasted work.

**Comment by jkbradley** [comment]
Just reviewed, and this LGTM  : )

**Comment by MLnick** [comment]
@jkbradley fair enough - I misinterpreted things thinking we were trying to nail down further Python parity. I do agree the focus should be on docs / QA. As it happens this was trivial to do while working on the doc update :)

File: python/pyspark/ml/feature.py (hunk 1 above)
Comment (MLnick) [inline python/pyspark/ml/feature.py]: @BryanCutler it's a good point - but I've seen that all other docs for the getters through py-ml omit the `:py:attr:`. I'm not certain if this is intended or some oversight that has become "copy-pasted" as de facto standard :) I for one would support updating all of them as it makes sense to me that the doc links match the setters. But it then needs to be done for all Python param setter methods to be consistent.

cc @jkbradley @holdenk @yanboliang

**Comment by MLnick** [comment]
jenkins retest this please

File: python/pyspark/ml/feature.py (hunk 1 above)
Comment (holdenk) [inline python/pyspark/ml/feature.py]: I think we should probably document the setters at some point, but perhaps 2.1 timeframe.

Comment (MLnick) [inline python/pyspark/ml/feature.py]: yeah - though it is a large, but actually trivial, change. We could probably do it in one pass right at the end.

**Pull request closed without merging**